cmd/clusterctl/client/repository/components.go

```diff
@@ -439,6 +446,32 @@ func fixCRDWebhookNamespaceReference(o unstructured.Unstructured, targetNamespac
 	return o, errors.Errorf("failed to patch %s CustomResourceDefinition", version)
 }
 
+// fixCertificate fixes the dnsNames of cert-manager Certificates.
+func fixCertificate(o unstructured.Unstructured, originalNamespace, targetNamespace string) (unstructured.Unstructured, error) {
+	dnsNames, ok, err := unstructured.NestedStringSlice(o.UnstructuredContent(), "spec", "dnsNames")
+	if err != nil {
+		return o, errors.Wrapf(err, "failed to get .spec.dnsNames from Certificate %s/%s", o.GetNamespace(), o.GetName())
+	}
+	// Return if we don't find .spec.dnsNames.
+	if !ok {
+		return o, nil
+	}
+
+	// Iterate through dnsNames and adjust the namespace.
+	// The dnsNames slice usually looks like this:
+	// - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc
+	// - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc.cluster.local
+	for i, dnsName := range dnsNames {
+		dnsNames[i] = strings.Replace(dnsName, originalNamespace, targetNamespace, 1)
```

Comment:
this is ok if we assume original namespace is never used as a substring in other places. To makes it bullet proof we need to prepend/append a dot

Reply:
Sounds good.

Reply:
Done, ptal